### PR TITLE
Faction Data: Add MUL ID

### LINKED ISSUE #4286: Missing Labels in the RAT Editor
This screenshot is from the RAT Gen Editor - Edit Factions Tab
image

The highlighted parts always show as 0,1,9 but I know from the XML files this is the part that you can edit the 'weightDistribution' element.

As a sample
<weightDistribution era='3100' unitType='Mek'>3,4,2,1</weightDistribution>  this is the 0
<weightDistribution era='3100' unitType='Tank'>3,4,3,1</weightDistribution> this is the 1
<weightDistribution era='3100' unitType='Aero'>1,1,1</weightDistribution> this is the 9

I can see code in the RATGEneratorEditor.java lines 1224-1225 that reference them, but I think they are missing the String Categories to be able to write the Mek, Tank, Aero versus 0,1,9. 

 Any chance we could get a fix to show the names instead of the 0,1,9 on the table.

### PR DESCRIPTION
**Pull request by SJuliez**
As the title says, this adds the faction ID from the MUL to the FactionRecord data, editable in the RATGeneratorEditor. I added faction IDs as far as I could match them.

Since the MUL is called from a few places, I unified the calling code. The whitespace changes in UIUtil weren't intentional.

Fixes #4286

### CODE REVIEW

**Comment by HammerGS** [comment]
Did some testing and the bug is fixed...Thanks! But nothing exported seems to have a Faction MUL ID in the csv file it still reads TBD.

**Comment by SJuliez** [comment]
Right, corrected.

File: megamek/src/megamek/client/ui/swing/util/UIUtil.java [hunk 1]
```diff
@@ -803,19 +826,19 @@ public void setToolTipText(String text) {
...
         }
     }
-    
-    /** 
-     * A MMComboBox with a specialized tooltip display. Displays the tooltip to the right side
 
-     * of the parent dialog, not following the mouse. 
+    /**
+     * A MMComboBox with a specialized tooltip display. Displays the tooltip to the right side
+     * <p>
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/util/UIUtil.java:832]: ?

**Review by NickAragua — APPROVED** [review]
Looks fine code-wise, except for that random <p> tag in one of the comments for no reason

**Comment by SJuliez** [comment]
Found out that I need to implement multiple MUL IDs that go with name changes within a single faction

**Comment by SJuliez** [comment]
Not happy with this

**Pull request closed without merging**